src/apple.rs

```diff
@@ -75,6 +75,58 @@ pub struct AppleResponse {
     /// An array that contains all in-app purchase transactions. This excludes transactions for consumable products
     /// that have been marked as finished by your app. Only returned for receipts that contain auto-renewable subscriptions.
     pub latest_receipt_info: Option<Vec<AppleLatestReceipt>>,
+    /// A JSON representation of the receipt that was sent for verification
+    pub receipt: Option<AppleReceipt>,
+}
+
+impl AppleResponse {
+    #[must_use]
+    /// Returns true if the receipt we are validating is from a subscription purchase
+    pub fn is_subscription(&self, transaction_id: &str) -> bool {
```

Comment:
It turns out that Apple gives us _all_ receipts associated with a user in the response, subscription or otherwise, making it much more complicated/costly to determine a) which package we're trying to validate without knowing the transaction id, and b) whether the package is a subscription (since it always sends subscription receipts, if a user has _ever_ subscribed, `latest_receipt_info` will be set even on package receipts.

Conveniently, we're always given the transaction id, and getting things by the transaction id raises the potential of a huge optimization in the subscription validation since we wouldn't have to iterate through to try and find the most recent subscription receipt